### PR TITLE
Fix sampling from DiscreteNonParametric with integer-valued probs

### PR DESCRIPTION
**Pull request by devmotion**
Fixes https://github.com/JuliaStats/Distributions.jl/issues/1111.

### CODE REVIEW

**Comment by devmotion** [comment]
Bump :slightly_smiling_face:

File: test/discretenonparametric.jl [hunk 1]
```diff
@@ -130,4 +130,11 @@ d = DiscreteNonParametric([0, 1, 2, 3, 4], [0.0f0, 0.1f0, 0.2f0, 0.3f0, 0.4f0])
 
 d = DiscreteNonParametric([4, 3, 2, 1, 0], [0.4f0, 0.3f0, 0.2f0, 0.1f0, 0.0f0])
 @test iszero(count(iszero(rand(d)) for _ in 1:100_000_000))
+
+# Sampling with integer-valued probabilities; see issue #1111
+d = DiscreteNonParametric([1, 2], [0, 1])
+@test iszero(count(isone(rand(d)) for _ in 1:100_000_000))
```
Comment (mschauer) [inline test/discretenonparametric.jl:136]: We are statisticians, we can take reasonably sized samples and still be reasonably sure 😉. Also time and energy consumption in mind

Comment (devmotion) [reply test/discretenonparametric.jl:136]: Haha yes, these numbers seem a bit large. I guess I just copied the tests above which require a larger sample size.

**Review by mschauer — APPROVED** [review]

**Comment by devmotion** [comment]
Thanks for merging this PR! Is it possible to make a new release with the fix?

**Pull request merged**